lang/test/org/partiql/lang/eval/evaluatortestframework/AstEvaluatorTestAdapter.kt

```diff
@@ -33,6 +31,89 @@ private fun EvaluatorTestDefinition.createPipeline(forcePermissiveMode: Boolean
     }
 }
 
+/** A generated and human readable description of this test case for display in assertion failure messages. */
+fun EvaluatorTestCase.testDetails(note: String, actualResult: String? = null): String {
+    val b = StringBuilder()
+    b.appendLine("Note            : $note")
+    b.appendLine("Group name      : $groupName")
+    b.appendLine("Query           : $query")
+    b.appendLine("Expected result : $expectedResult")
+    if (actualResult != null) {
+        b.appendLine("Actual result   : $actualResult")
+    }
+    b.appendLine("Result format   : $expectedResultFormat")
+
+    return b.toString()
+}
+
+/** A generated and human readable description of this test case for display in assertion failure messages. */
+fun EvaluatorErrorTestCase.testDetails(
+    note: String,
+    actualErrorCode: String? = null,
```

Comment:
For consistency w/ `actualErrorContext`, could this argument have type `ErrorCode?` rather than `String?`?

Reply:
sure.